selfservice/strategy/code/code_recovery_test.go

```diff
@@ -0,0 +1,91 @@
+package code_test
+
+import (
+	"database/sql"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/ory/kratos/internal"
+	"github.com/ory/kratos/selfservice/strategy/code"
+	"github.com/ory/kratos/x"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ory/x/stringslice"
+	"github.com/ory/x/urlx"
+
+	"github.com/ory/kratos/selfservice/flow"
+	"github.com/ory/kratos/selfservice/flow/recovery"
+)
+
+func TestRecoveryCode(t *testing.T) {
+	conf, _ := internal.NewFastRegistryWithMocks(t)
+
+	newCode := func(expiresIn time.Duration, f *recovery.Flow) *code.RecoveryCode {
+		return &code.RecoveryCode{
+			ID:        x.NewUUID(),
+			FlowID:    f.ID,
+			ExpiresAt: time.Now().Add(expiresIn),
+		}
+	}
+
+	req := &http.Request{URL: urlx.ParseOrPanic("https://www.ory.sh/")}
+	t.Run("func=GenerateRecoveryCode", func(t *testing.T) {
+		t.Run("case=creates unique tokens", func(t *testing.T) {
+			codes := make([]string, 100)
```

Comment:
The chance that two codes are the same is `100/10^8 = 1/10^6` which should be low enough to not cause flakes here. If my statistics are correct, and I am bad in statistics :joy:

Reply:
Should be fine ™️